File: src/main/java/com/example/the_machine/operator/result/OperatorResultCode.java
package com.example.the_machine.operator.result;

import com.fasterxml.jackson.annotation.JsonValue;

/**
 * Enum representing the common result codes for operator execution. Provides type-safe constants
 * for the most frequently used codes.
 */
public enum OperatorResultCode {
  SUCCESS, VALIDATION_ERROR, EXCEPTION;

  @JsonValue
  public String getValue() {
    return name();
  }
}

File: src/main/kotlin/com/example/the_machine/operator/result/OperatorResultCode.kt
package com.example.the_machine.operator.result

import kotlinx.serialization.Serializable

/**
 * Enum representing the common result codes for operator execution. Provides type-safe constants
 * for the most frequently used codes.
 */
@Serializable
enum class OperatorResultCode {

  SUCCESS,
  VALIDATION_ERROR,
  EXECUTION_FAILED,
  UNEXPECTED_ERROR,
  RETRY_LOOP_ERROR,
  BUDGET_EXCEEDED;

}